### PR TITLE
WIP: Use websocket for notification instead of eventsource

### PR DESCRIPTION
**Pull request by lunny**

### CODE REVIEW

File: web_src/js/features/notification.js [hunk 1]
```diff
@@ -7,6 +7,13 @@ export function initNotificationsTable() {
   const table = document.getElementById('notification_table');
   if (!table) return;
 
+  const url = 'ws://' + window.location.host + '/ws';
+  const ws = new WebSocket(url);
```
Comment (silverwind) [inline web_src/js/features/notification.js:11]:
```suggestion
  const ws = new WebSocket(`${window.location.origin.replace(/^http/, 'ws')}/ws`);
```

So it works on HTTPS as well. Should move this code to a separate `modules/websocket.js` file later.

**Comment by lunny** [comment]
> Good choice with melody, it's something I've been eyeing as well because of easy broadcasts.

I just found melody has a memory switch system so I may change it back to gorilla/webhook directly.

**Comment by silverwind** [comment]
> I just found melody has a memory switch system

What do you mean? It seems pretty bare-bone: https://github.com/olahol/melody/blob/master/melody.go

**Comment by yardenshoham** [comment]
What are the implications for the helm chart?

cc @pat-s

**Comment by silverwind** [comment]
Reverse proxies may need adjustment to pass through Websockets. For nginx specifically this is it:

```nginx
proxy_set_header Upgrade $http_upgrade;
proxy_set_header Connection $http_connection;
````

**Comment by pat-s** [comment]
@lunny Do you plans to continue here?

**Comment by lunny** [comment]
> @lunny Do you plans to continue here?

Yes. I will send some code recently. We need a pub/sub interfaces/services on server side.

**Comment by lunny** [comment]
Close as it is outdated.

**Pull request closed without merging**